File: Samples-Java/Samples/src/main/java/fragments/dataform/DataFormEditorsFragment.java
package fragments.dataform;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import com.telerik.android.common.Function;
import com.telerik.android.sdk.R;
import com.telerik.widget.dataform.engine.EntityProperty;
import com.telerik.widget.dataform.engine.PropertyChangedListener;
import com.telerik.widget.dataform.visualization.RadDataForm;
import com.telerik.widget.dataform.visualization.core.EntityPropertyEditor;

import activities.ExampleFragment;

public class DataFormEditorsFragment extends Fragment implements ExampleFragment {
    @Override
    public String title() {
        return "Editors";
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        ViewGroup rootLayout = (ViewGroup)inflater.inflate(R.layout.fragment_dataform_editors, null);

        RadDataForm dataForm = new RadDataForm(this.getActivity());
        dataForm.getAdapter().setEditorProvider(new Function<EntityProperty, EntityPropertyEditor>() {
            @Override
            public EntityPropertyEditor apply(EntityProperty property) {
                if(property.name().equals("EmployeeType")) {
                    return new CustomEditor(getActivity(), property);
                }

                return null;
            }
        });

        dataForm.setEntity(new Person());

        rootLayout.addView(dataForm);

        return rootLayout;
    }

    public class CustomEditor extends EntityPropertyEditor implements View.OnClickListener, PropertyChangedListener {
        private EmployeeType type;
        private Button editorButton;

        public CustomEditor(Context context, EntityProperty property) {
            super(context, R.layout.dataform_custom_editor, R.id.custom_editor_header, R.id.custom_editor, R.id.custom_editor_validation_view, property);

            editorButton = (Button)editorView;
            editorButton.setOnClickListener(this);

            ((TextView)headerView).setText(property.getHeader());
        }

        @Override
        public Object value() {
            return type;
        }

        @Override
        protected void applyEntityValueToEditor(Object o) {
            if(o == null) {
                this.editorButton.setText("Tap to select.");
                return;
            }

            this.editorButton.setText(o.toString());
            type = (EmployeeType)o;
        }

        @Override
        public void onClick(View v) {
            this.showEditorFragment(this.type);
        }

        private void showEditorFragment(EmployeeType initialValue) {
            CustomEditorFragment fragment = new CustomEditorFragment();
            fragment.addPropertyChangedListener(this);
            fragment.setType(initialValue);
            fragment.show(getActivity().getSupportFragmentManager(), "customEditor");
        }

        @Override
        protected boolean supportsType(Class type) {
            // Defining which types this editor supports is required.
            return type == EmployeeType.class;
        }

        @Override
        public void onPropertyChanged(String s, Object o) {
            if(o == type) {
                return;
            }

            applyEntityValueToEditor(o);

            // Remember to call value changed, otherwise the object being edited will not be updated.
            onEditorValueChanged(o);
        }
    }
}

File: Samples-Xamarin/Fragments/Dataform/DataFormEditorsFragment.cs
using System;
using Android.App;
using Android.OS;
using Android.Views;
using Android.Widget;
using Com.Telerik.Widget.Dataform.Engine;
using Com.Telerik.Widget.Dataform.Visualization;
using Com.Telerik.Widget.Dataform.Visualization.Core;
using Com.Telerik.Android.Common;

namespace Samples
{
	public class DataFormEditorsFragment : Android.Support.V4.App.Fragment, ExampleFragment, IFunction
	{
		RadDataForm dataForm;
		public override View OnCreateView (LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
		{
			ViewGroup rootLayout = (ViewGroup)inflater.Inflate(Resource.Layout.fragment_dataform_editors, container, false);

			dataForm = (RadDataForm)rootLayout.FindViewById(Resource.Id.data_form_editors);
			dataForm.Adapter.SetEditorProvider(this);

			dataForm.SetEntity (new Person ());

			return rootLayout;
		}

		public Java.Lang.Object Apply (Java.Lang.Object p0)
		{
			IEntityProperty property = (IEntityProperty)p0;
			if(property.Name().Equals("EmployeeType")) {
				return new CustomEditor(dataForm, property, this.Activity.FragmentManager);
			}

			return null;
		}

		public String Title()
		{
			return "Editors";
		}

		public class CustomEditor : EntityPropertyEditor, View.IOnClickListener, IPropertyChangedListener {
			private EmployeeType type;
			private Button editorButton;
			private FragmentManager fragmentManager;

			public CustomEditor(RadDataForm dataForm, IEntityProperty property, FragmentManager fragmentManager) : base(dataForm,
					dataForm.EditorsMainLayout,
					dataForm.EditorsHeaderLayout,
					Resource.Id.data_form_text_viewer_header,
					Resource.Layout.dataform_custom_editor,
					Resource.Id.custom_editor,
					dataForm.EditorsValidationLayout,
					property)
			{

				editorButton = (Button)EditorView;
				editorButton.SetOnClickListener(this);

				((TextView)HeaderView).Text = property.Header;
				this.fragmentManager = fragmentManager;
			}

			public override Java.Lang.Object Value()
			{
				return type.ToString();
			}

			protected override void ApplyEntityValueToEditor(Java.Lang.Object o)
			{
				if(o == null) {
					this.editorButton.Text = "Tap to select.";
					return;
				}

				this.editorButton.Text = o.ToString();
				//type = (EmployeeType)Enum.Parse(typeof(EmployeeType), o.ToString());
				type = (EmployeeType)Enum.Parse(type.GetType(), o.ToString());
			}

			public void OnClick(View v)
			{
				this.ShowEditorFragment(this.type);
			}

			private void ShowEditorFragment(EmployeeType initialValue)
			{
				CustomEditorFragment fragment = new CustomEditorFragment();
				fragment.AddPropertyChangedListener(this);
				fragment.Type = initialValue;
				fragment.Show(this.fragmentManager, "customEditor");
			}

			public void OnPropertyChanged(String s, Java.Lang.Object o)
			{
				if(o.ToString() == type.ToString()) {
					return;
				}

				ApplyEntityValueToEditor(o);

				// Remember to call value changed, otherwise the object being edited will not be updated.
				OnEditorValueChanged(o);
			}
		}
	}
}

